todo_app/app.py

```diff
@@ -1,44 +1,69 @@
+from datetime import date
+from os import stat
 from flask import Flask, render_template, redirect, request
 
-from todo_app.data.session_items import *
+from todo_app.data.trello_items import *
 from todo_app.flask_config import Config
 
 app = Flask(__name__)
 app.config.from_object(Config())
 
 
 @app.route('/')
-def index():    
-    list = sorted(get_items(), key=lambda item:item['status'], reverse=True)
-    return render_template("index.html", to_do_items=list)
+def index():
+    sorted_list = sorted(get_tasks(), key=lambda item:item.status, reverse=True)
+    number_of_tasks = sorted_list.count
+    return render_template("index.html", to_do_items=sorted_list, empty_to_do_list=number_of_tasks)
+
+@app.route('/<status>')
+def filtered_index(status):
+    status_filter = ''
+    match(status):
+        case 'not-started':
+           status_filter = 'Not Started'
+        case 'in-progress':
+            status_filter = 'In Progress'
+        case 'complete':
+            status_filter = 'Complete' 
+    filtered_list = get_filtered_tasks(status_filter)
+    number_of_tasks = bool(len(filtered_list) == 0)
+    return render_template("index.html", to_do_items=filtered_list, empty_to_do_list=number_of_tasks)
 
 @app.route('/', methods=['POST'])
 def add_to_do():
-    add_item(request.form.get('to-do-title'), request.form.get('to-do-notes'))
+    due_date = request.form.get('task-due-date')
+    if due_date != "":
+        due_date = datetime.strptime(request.form.get('task-due-date'), "%d/%m/%Y")
+    create_task(request.form.get('to-do-title'), due_date, request.form.get('to-do-notes'))
     return redirect('/')
 
 @app.route('/task/<id>')
 def view_to_do_item(id):
-    task = get_item(id)
+    task = get_task(id)
     return render_template("task.html", task=task)
 
 @app.route('/task/<id>', methods=['POST'])
 def amend_item(id):
     amended_status = request.form.get('task-status')
     amended_title = request.form.get('task-title')
     amended_notes = request.form.get('task-notes')
-    task = { 'id': int(id), 'status': amended_status, 'title': amended_title, 'notes': amended_notes }
-    save_item(task)
-    return redirect('/')
+    amended_due_complete = "true" if amended_status == "Complete" else "false"
+    amended_due_date = datetime.strptime(request.form.get('task-due-date'), "%d/%m/%Y")
```

Comment:
spotted a bug here! If you try to update a task that doesn't have a due date it will throw an error: 
`ValueError: time data '' does not match format '%d/%m/%Y'`
`File "\DevOps-Course-Starter\todo_app\app.py", line 51, in amend_item`

I expect you either want to:
- sensibly handle not having this due date field present or
- make the due date a required field.